auth/src/main/java/com/firebase/ui/auth/ui/email/EmailActivity.java

```diff
@@ -173,11 +173,14 @@ public void onNewUser(User user) {
     public void onTroubleSigningIn(String email) {
         TroubleSigningInFragment troubleSigningInFragment = TroubleSigningInFragment.newInstance
                 (email);
-        switchFragment(troubleSigningInFragment, TroubleSigningInFragment.TAG, true);
+        switchFragment(troubleSigningInFragment, TroubleSigningInFragment.TAG, true, true);
     }
 
     @Override
     public void onClickResendEmail(String email) {
+        if (getSupportFragmentManager().getBackStackEntryCount() > 0) {
```

Comment:
Can you just add a comment here detailing the assumption you're making?  As I understand it, you're assuming that if there's any back stack at all here it's due to the trouble sending flow and you want to pop from the stack.

Reply:
Good idea. Done. :smiley: